.github/scripts/detectRedirectCycle.ts

```diff
@@ -0,0 +1,64 @@
+import {parse} from 'csv-parse';
+import fs from 'fs';
+
+const parser = parse();
+
+const adjacencyList: Record<string, string[]> = {};
+const visited: Map<string, boolean> = new Map<string, boolean>();
+const backEdges: Map<string, boolean> = new Map<string, boolean>();
+
+function addEdge(source: string, target: string) {
+    if (!adjacencyList[source]) {
+        adjacencyList[source] = [];
+    }
+    adjacencyList[source].push(target);
+}
+
+function isCyclic(currentNode: string): boolean {
+    visited.set(currentNode, true);
+    backEdges.set(currentNode, true);
+
+    // Do a depth first search for all the neighbours. If a node is found in backedge, a cycle is detected.
+    const neighbours = adjacencyList[currentNode];
+    if (neighbours) {
```

Comment:
Looks like you could safely remove this if statement

Reply:
No, there will be nodes that have no neighbours because this is a directed graph.

eg: 
```
urlA, urlB
```

Graph will look like this
`urlA -> urlB`

`urlB` has no neighbours that we can go to. So we'll try to iterate over undefined in next line